app/models/time_interval.rb

```diff
@@ -0,0 +1,20 @@
+##
+# An interval of time between two concrete points in time
+class TimeInterval
+  ##
+  # Creates a TimeInterval stargin at the given point in time with the given
+  # duration expressed as a number of seconds
+  def initialize(starting_at:, duration:)
+    @starting_at = starting_at
+    @duration = duration
+    @closed_end_at = @starting_at + @duration - 0.01.seconds
```

Comment:
minor question: why not just `@end_at = @starting_at + @duration` and then in any place we could use `<` or `<=` ?